src/billo.php

```diff
@@ -32,6 +32,18 @@ function distributeMessage($listsConfig, $mailSender, $id, $message) {
     $mailSender->Subject =
       '['.$listsConfig->getListName($list->getEmail()).'] '.$message->subject;
 
+    $members = $listsConfig->getMembers($list->getEmail());                                                                                                                                         
+     // Check at least one of the senders ("from") is subscribed to the list                                                                     
+     $subscribed = false;                                                                                                                        
+     foreach($message->getHeader('from')->getAddressList() as $from) {
+      if ($listsConfig->isMember($from->getEmail(), $members)) {
+        $subscribed = true;                                
+       }                                
+     }                                                                                                                                           
+     if(!$subscribed) {
+      echo "Not distributed message from unknown sender" . PHP_EOL;
+       continue;                               
+     }
```

Comment:
We could move this logic to the ListsConfigParser ... naming the function is probably the hardest part :-D 

Suggestion:
```php
if(!$listsConfig->isAnyAddressMemberOfList($message->getHeader('from')->getAddressList(), $list->getEmail())) {
  continue;
}
```

Keeps the complexity of this function lower and gives us a chance to unit test it...